assets/assets/utils/common_utils.py

```diff
@@ -58,15 +63,22 @@ def get_mimetype(file: bytes) -> Any:
     return mimetype
 
 
-def get_pages(file: bytes) -> Any:
-    pages = get_pages_from_pdf(file) or get_pages_from_image(file)
-    return pages
+def get_pages(file: bytes, ext: str) -> Any:
+    logger_.info("Getting count of pages by file extension: %s", ext)
+    if ext not in chem_utils.SUPPORTED_FORMATS:
+        raise exceptions.AssetsUnsupportedFileFormat(
+            "File extension %s is not supported", ext
+        )
+    if "chem" == chem_utils.SUPPORTED_FORMATS[ext]:
+        return chem_utils.get_page_count(file, ext)
+    # Pdf as default
+    return get_pages_from_pdf(file) or get_pages_from_image(file)
 
 
 def get_pages_from_pdf(file: bytes) -> Any:
     try:
         pages = pdf2image.pdfinfo_from_bytes(file)["Pages"]
-    except:  # noqa
+    except Exception:  # noqa
```

Comment:
Do we need `# noqa` after clarification of exception? Flake8 shouldn't complain about this anymore, but need to check